File: callie-scala/src/org/callie/math/Axis.scala
package org.callie.math

object Axis extends Enumeration{
  case class AxisValue(axis:Value, value:Float)

  val X, Y, Z = Value

  def apply(s:String) = s match {
    case "x" => X
    case "y" => Y
    case "z" => Z
  }
}

File: callie-scala/src/main/scala/org/callie/math/Axis.java
package org.callie.math;

public enum  Axis {
    X, Y, Z;

    public static Axis apply(String s) {
        if("x".equalsIgnoreCase(s)) return X;
        if("y".equalsIgnoreCase(s)) return Y;
        if("z".equalsIgnoreCase(s)) return Z;
        throw new IllegalArgumentException(s);
    }
}
